pilot/pkg/security/authz/builder/builder_test.go

```diff
@@ -189,63 +181,57 @@ func TestBuilder_BuildTCPFilter(t *testing.T) {
 				policy.SimpleRole("role-1", "a", "foo"),
 				policy.SimpleBinding("binding-1", "a", "role-1"),
 			},
-			wantRules:            true,
+			wantRules:            false,
```

Comment:
Why are these changed from `true` to `false` in the unit tests?

Reply:
This is because {a: {b: []}} is marshalled differently between struct roundtripping and directly bytes. What does this test actually test? I couldn't tell and I've been asking around.

Reply:
This is testing the [policies_](https://github.com/envoyproxy/envoy/blob/4825facf2451b234af737b5af8c792c31bca5d41/source/extensions/filters/common/rbac/engine_impl.cc#L34) with empty rules.

Does `Any` changes the semantics in the unmarshalled protobuf in Envoy?

Reply:
I don't know - that's why I need you to double-check :) Envoy is removing struct support in the next version, so we should stop generating the structs if we want to be compatible with the latest envoy.

Reply:
Any follow up on this? It would be nice to just user Any in master